diff --git a/TestCoverageConsole/TestDiff.cpp b/TestCoverageConsole/Tes
index 8944117..c9f2773 100644
--- a/TestCoverageConsole/TestDiff.cpp
+++ b/TestCoverageConsole/TestDiff.cpp
@@ -24,10 +24,10 @@ namespace TestCoverageConsole
        void FilterByDiff()
        {
                int value1 = 0;
-               if (true)
-                       int value2 = 0;
-               if (false)
-                       int value3 = 0;
+               if (true)                       // In Diff
+                       int value2 = 0; // In Diff
+               if (false)                      // In Diff
+                       int value3 = 0; // In Diff
                int value4 = 0;
        }
 }
\ No newline at end of file